### PR TITLE
fixed issue 10873

### PR DESCRIPTION
**Pull request by GhastCraftHD**
Fix to issue [10873](https://github.com/PaperMC/Paper/issues/10873)
Correctly cleared enchantment map for ItemMeta#removeEnchantments

### CODE REVIEW

File: patches/server/1037-General-ItemMeta-fixes.patch [hunk 1]
```diff
@@ -906,6 +908,15 @@ index 705d7365a7bab9fe29b90a175040e6a1623e21a7..8ccb8d71ee489891e8d9128a5520675d
          this.lore = lore != null ? io.papermc.paper.adventure.PaperAdventure.asVanilla(lore) : null;
      }
      // Paper end
+@@ -1094,7 +1099,7 @@ class CraftMetaItem implements ItemMeta, Damageable, Repairable, BlockDataMeta {
+     @Override
+     public void removeEnchantments() {
+         if (this.hasEnchants()) {
+-            this.enchantments.clear();
++            this.enchantments = null;
```
Comment (kennytv) [inline patches/server/1037-General-ItemMeta-fixes.patch]: needs a paper comment, `// Paper - Correctly clear enchantments`

Comment (GhastCraftHD) [reply patches/server/1037-General-ItemMeta-fixes.patch]: added comment

Comment (kennytv) [reply patches/server/1037-General-ItemMeta-fixes.patch]: Should be on the same line, after the ;

Comment (GhastCraftHD) [reply patches/server/1037-General-ItemMeta-fixes.patch]: fixed ._.

**Review by kennytv — APPROVED** [review]
Thank you!

**Pull request merged**